sqlmesh/core/model/meta.py

```diff
@@ -158,6 +162,46 @@ def _columns_validator(
 
         return v
 
+    @validator("table_properties_", pre=True)
+    def _properties_validator(
+        cls, v: t.Any, values: t.Dict[str, t.Any]
+    ) -> t.Optional[t.Dict[str, TableProperty]]:
+        if v is None or not isinstance(v, exp.Expression):
+            return v
+
+        dialect = values.get("dialect")
+
+        if isinstance(v, (exp.Tuple, exp.Array)):
+            eq_expressions: t.List[exp.Expression] = v.expressions
+        elif isinstance(v, exp.Paren):
+            eq_expressions = [v.this]
+        else:
+            eq_expressions = [v]
+
+        table_properties = {}
+        for eq_expr in eq_expressions:
+            if not isinstance(eq_expr, exp.EQ):
+                raise ConfigError(
+                    f"Invalid table property '{eq_expr.sql(dialect=dialect)}'. "
+                    "Table properties must be specified as key-value pairs <key> = <value>. "
+                )
+
+            key = eq_expr.this.name
+            value_expr = eq_expr.expression
+            if isinstance(value_expr, exp.Boolean) or value_expr.is_string:
+                value = value_expr.this
+            elif value_expr.is_int:
+                value = int(value_expr.this)
+            elif value_expr.is_number:
+                value = float(value_expr.this)
+            else:
+                value = value_expr.copy()
```

Comment:
you're expecting an expression here?

do you want to use the python evaluator here to convert sql into python?

Reply:
No, not really. I think I do want to preserve an expression. If someone does something like `TIMESTAMP('<timestamp>')` I want to pass it as is, don't I?

Reply:
then why are you converting anything at all?

Reply:
To be consistent with scenario when the dictionary is constructed and passed directly from Python or dbt models.

Reply:
i don't think it makes a difference because you call convert anyways

Reply:
or just call convert in the validator, so when python models pass it in, it is always an expression type

Reply:
updated